test/e2e/storage/testsuites/testdriver.go

```diff
@@ -78,14 +78,17 @@ type DynamicPVTestDriver interface {
 	GetClaimSize() string
 }
 
-// Capability represents a feature that a volume plugin supports
+// Capability represents a feature that a volume plugin supports.
+// Unless noted otherwise, capabilities are assumed to be not
+// supported when not set explicitly.
 type Capability string
 
 const (
 	CapPersistence Capability = "persistence" // data is persisted across pod restarts
 	CapBlock       Capability = "block"       // raw block mode
 	CapFsGroup     Capability = "fsGroup"     // volume ownership via fsGroup
 	CapExec        Capability = "exec"        // exec a file in the volume
+	CapMultiPODs   Capability = "multipods"   // multiple pods on a node can use the same volume concurrently; enabled by default
```

Comment:
how/where is this enabled by default?

Reply:
At the place where it is getting checked. If unset, the code assumes that the driver supports multiple pods:
https://github.com/kubernetes/kubernetes/blob/5551371f0dde5ec7be93659d2cce4d92fa3f6eca/test/e2e/storage/testsuites/provisioning.go#L201-L205

But I am no longer sure what the correct behavior is in this situation. Technically it can work and I thought `NodeStageVolume` was meant to enable this double use on the same node, but the CSI spec PR makes it clear that this usage was not meant to be supported.

Reply:
I don't think the spec being insufficient means this test is invalid or that this test is an endorsement of unsupported behaviour. My understanding is that this behaviour (ability for multiple pods on 1 node to write to the same volume) is not explicitly called out in kubernetes docs but it follows naturally from the way kubernetes does mounts and defines access modes (RWO) in terms of nodes, not pods. So many users rely on it.

We know that the GCE PD CSI driver supports this capability even though it reports SINGLE_WRITER, we know SINGLE_WRITER isn't enough to capture the driver's capabilities...rather than wait for the spec to update to accurately capture what kubernetes has allowed plugins/drivers/users to do since before CSI, we can add the test now and set the capability true for drivers as we see fit (this means disabling this bool by default like the others, which we should do anyway IMO for consistency).

Seems the latest discussion on the spec issue is here: https://github.com/container-storage-interface/spec/issues/178.

Reply:
I agree that this spec limitation seems rather artificial and I am fine with having a test that explicitly tests support for this particular even if it is not covered by the spec (due to #178 being open).